### PR TITLE
Support parsing generated columns in CREATE TABLE

### PR DESCRIPTION
**Pull request by tyamagu2**
Support parsing [generated columns](https://cloud.google.com/spanner/docs/generated-column/how-to) in CREATE TABLE.

Note that the modified parser parses generated column expression as an ordinary expression, and thus it accepts expressions that are invalid as generated column expression as long as `parseExpr` is able to parse them.

### CODE REVIEW

File: pkg/ast/ast.go [hunk 1]
```diff
@@ -1272,10 +1272,24 @@ type ColumnDef struct {
...
 	Null token.Pos // position of "NULL"
 
-	Name    *Ident
-	Type    SchemaType
-	NotNull bool
-	Options *ColumnDefOptions // optional
+	Name          *Ident
+	Type          SchemaType
+	NotNull       bool
+	GeneratedExpr *GeneratedColumnExpr // optional
```
Comment (MakeNowJust) [inline pkg/ast/ast.go:1278]: Please update the above `ColumnDef` doc comment by adding `{{ GeneratedExpr | sqlOpt }}`.

File: pkg/ast/ast.go [hunk 2]
```diff
@@ -1272,10 +1272,24 @@ type ColumnDef struct {
...
+	Options       *ColumnDefOptions    // optional
+}
+
+// GeneratedColumnExpr is generated column expression.
+//
+//     AS ({{.Expr | sql}}) STORED
+type GeneratedColumnExpr struct {
```
Comment (MakeNowJust) [inline pkg/ast/ast.go:1285]: Are `GeneratedColumnExpr.Pos` and `GeneratedColumnExpr.End` missing? They are necessary to implement `Node` interface.

**Review by MakeNowJust — CHANGES_REQUESTED** [review]
Great :tada:

Thanks for noting about parsing. I think such validation is done in analyzing phase instead of parsing, so that's no problem.

File: pkg/ast/ast.go (hunk 2 above)
Comment (tyamagu2) [reply pkg/ast/ast.go:1286]: Added `Pos`, `End` and `SQL` to `GeneratedColumnExpr`. Also fixed `ColumnDef.End` to take `GeneratedExpr` into account.

**Review by MakeNowJust — APPROVED** [review]

**Comment by MakeNowJust** [comment]
Thank you @tyamagu2!

**Pull request merged**